apps/examples/testcase/le_tc/kernel/itc_pthread.c

```diff
@@ -0,0 +1,522 @@
+	TC_ASSERT_EQ("pthread_cond_destroy", ret_chk, EINVAL);
+
+	ret_chk = pthread_condattr_destroy(&attr);
+	TC_ASSERT_EQ("pthread_condattr_destroy", ret_chk, OK);
+
+	TC_SUCCESS_RESULT();
+}
+
+/**
+* @fn                   :itc_pthread_pthread_cond_destroy_n_redestroy
+* @brief                :pthread_cond_init initialises the condition variable referenced by cond with attributes referenced by attr
+*                        pthread_cond_destroy destroy the given condition variable specified by cond
+* @Scenario             :pthread_cond_destroy re-destroying the object as already being destroyed
+* API's covered         :pthread_cond_init, pthread_cond_destroy
+* Preconditions         :none
+* Postconditions        :none
+*/
+static void itc_pthread_pthread_cond_destroy_n_redestroy(void)
+{
+	int ret_chk;
+	pthread_condattr_t attr;
+	pthread_cond_t cond_attrparam;
+
+	ret_chk = pthread_condattr_init(&attr);
+	TC_ASSERT_EQ("pthread_condattr_init", ret_chk, OK);
+
+	ret_chk = pthread_cond_init(&cond_attrparam, &attr);
+	TC_ASSERT_EQ("pthread_cond_init", ret_chk, OK);
+
+	ret_chk = pthread_cond_destroy(&cond_attrparam);
+	TC_ASSERT_EQ("pthread_cond_destroy", ret_chk, OK);
+
+	ret_chk = pthread_cond_destroy(&cond_attrparam);
+	TC_ASSERT_EQ("pthread_cond_destroy", ret_chk, EINVAL);
+
+	ret_chk = pthread_condattr_destroy(&attr);
+	TC_ASSERT_EQ("pthread_condattr_destroy", ret_chk, OK);
+
+	TC_SUCCESS_RESULT();
+}
+
+/**
+* @fn                   :itc_pthread_pthread_cond_broadcast_n_after_destroy
+* @brief                :broadcast or signal a condition
+* @Scenario             :pthread_cond_broadcast when object as already being destroyed.
+* API's covered         :pthread_cond_broadcast ,pthread_cond_init , pthread_cond_destroy , pthread_cond_wait
+* Preconditions         :none
+* Postconditions        :none
+*/
+static void itc_pthread_pthread_cond_broadcast_n_after_destroy(void)
+{
+	int ret_chk;
+	pthread_cond_t cond_attrparam;
+	pthread_condattr_t attr;
+
+	ret_chk = pthread_cond_init(&cond_attrparam, &attr);
+	TC_ASSERT_EQ("pthread_cond_init", ret_chk, OK);
+
+	ret_chk = pthread_cond_destroy(&cond_attrparam);
+	TC_ASSERT_EQ("pthread_cond_destroy", ret_chk, EINVAL);
+
+	ret_chk = pthread_cond_broadcast(&cond_attrparam);
+	TC_ASSERT_EQ("pthread_cond_broadcast", ret_chk, EINVAL);
+
+	TC_SUCCESS_RESULT();
+}
+
+/**
+* @fn                   :itc_pthread_setgetname_np_p_reset_name
+* @brief                :The pthread_getname_np function gets the name of pthread
+* @Scenario             :create pthread and get its default name
+* API's covered         :pthread_setname_np, pthread_getname_np
+* Preconditions         :none
+* Postconditions        :none
+*/
+static void itc_pthread_setgetname_np_p_reset_name(void)
+{
+	int ret_chk;
+	int exec_index;
+	int len;
+	pthread_t name_th;
+	char thread_name[BUF_LEN];
+	char *thread_str = "NameThread";
+	char get_name[BUF_LEN];
+	g_exec_index = 0;
+
+	len = strlen(thread_name);
```

Comment:
for strlen, #include <string.h> is needed

Reply:
added #include <string.h> in itc_pthread.c file